src/portalnet/protocol.rs

```diff
@@ -52,22 +54,30 @@ impl TalkReqHandler for ProtocolHandler {
 
 impl PortalnetProtocol {
     pub async fn new(portal_config: PortalnetConfig) -> Result<Self, String> {
+        let listen_all_ips = SocketAddr::new("0.0.0.0".parse().unwrap(), portal_config.listen_port);
+
+        let external_addr = match socket::stun_for_external(&listen_all_ips) {
+            Some(addr) => addr,
+            None => socket::default_local_address(portal_config.listen_port),
+        };
```

Comment:
[`unwrap_or_default`](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap_or_default)

Reply:
Hm, this method might be better called `discover_local_address`. If I understand `unwrap_or_default` correctly, I wouldn't be able to pass in any variable (like the port to listen on) to the default definition. (and would rust even let me define a Default for a `SocketAddr`, since it's in an external module?)

Reply:
ah yes, of course